### PR TITLE
#22617 fix ExpressionChangedAfterItHasBeenCheckedError in DotDeviceSelectorComponent

### PR DESCRIPTION
**Pull request by oidacra**
68747470733a2f2f696d616765732e7a656e68756275736572636f6e74656e742e636f6d2f3632316533623264333535646337333735373433356164362f34396337333964632d616232642d343562642d383763362d616137316137633161666432

### CODE REVIEW

File: core-web/apps/dotcms-ui/src/app/view/components/dot-device-selector/dot-device-selector.component.ts [hunk 1]
```diff
@@ -28,9 +32,15 @@ export class DotDeviceSelectorComponent implements OnInit, OnChanges {
...
     constructor(
         private dotDevicesService: DotDevicesService,
-        private dotMessageService: DotMessageService
+        private dotMessageService: DotMessageService,
+        private readonly cd: ChangeDetectorRef
     ) {}
 
+    // Due the @input value change before change detection has been run/or completed.
+    ngAfterViewChecked() {
+        this.cd.detectChanges();
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/view/components/dot-device-selector/dot-device-selector.component.ts:41]: Can we have a test to assert that this runs correctly?

**Review by fmontes — CHANGES_REQUESTED** [review]

**Review by rjvelazco — APPROVED** [review]

File: core-web/apps/dotcms-ui/src/app/view/components/dot-device-selector/dot-device-selector.component.ts (hunk 1 above)
Comment (oidacra) [reply core-web/apps/dotcms-ui/src/app/view/components/dot-device-selector/dot-device-selector.component.ts]: I removed these lines, I triggered the detectChanges at lines 79 and 84

**Review by alfredo-dotcms — APPROVED** [review]

File: core-web/apps/dotcms-ui/src/app/view/components/dot-device-selector/dot-device-selector.component.ts [hunk 2]
```diff
@@ -72,10 +76,12 @@ export class DotDeviceSelectorComponent implements OnInit, OnChanges {
                     if (this.disabled) {
                         this.placeholder = 'No devices';
                     }
+                    this.cd.detectChanges();
                 },
                 () => {
                     this.disabled = true;
                     this.placeholder = 'No devices';
+                    this.cd.detectChanges();
```
Comment (rjvelazco) [inline core-web/apps/dotcms-ui/src/app/view/components/dot-device-selector/dot-device-selector.component.ts:84]: We can add a third callback (complete), so we can add `this.cd.detectChanges()` once.

```suggestion
      },
      () => {
          this.disabled = true;
          this.placeholder = 'No devices';
       },
      () =>  this.cd.detectChanges()
   )
```

**Review by rjvelazco — APPROVED** [review]

File: core-web/apps/dotcms-ui/src/app/view/components/dot-device-selector/dot-device-selector.component.ts (hunk 2 above)
Comment (oidacra) [reply core-web/apps/dotcms-ui/src/app/view/components/dot-device-selector/dot-device-selector.component.ts:84]: added

**Review by fmontes — APPROVED** [review]

**Pull request merged**